src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -159,6 +159,53 @@ public long getLongMetric(String metric)
   }
 
   @Override
+  public int getIntMetric(String metric)
+  {
+    Object metricValue = event.get(metric);
+
+    if (metricValue == null) {
+      return 0;
+    }
+
+    if (metricValue instanceof Number) {
+      return ((Number) metricValue).intValue();
+    } else if (metricValue instanceof String) {
+      try {
+        String s = ((String) metricValue).replace(",", "");
+        return LONG_PAT.matcher(s).matches() ? Integer.valueOf(s) : Double.valueOf(s).intValue();
+      }
+      catch (Exception e) {
+        throw new ParseException(e, "Unable to parse metrics[%s], value[%s]", metric, metricValue);
+      }
+    } else {
+      throw new ParseException("Unknown type[%s]", metricValue.getClass());
+    }
+  }
+
+  @Override
+  public double getDoubleMetric(String metric)
+  {
+    Object metricValue = event.get(metric);
+
+    if (metricValue == null) {
+      return 0.0d;
+    }
+
+    if (metricValue instanceof Number) {
+      return ((Number) metricValue).doubleValue();
+    } else if (metricValue instanceof String) {
+      try {
+        return Double.valueOf(((String) metricValue).replace(",", ""));
```

Comment:
can we create a method that does `((String) metricValue).replace(",", "");`, it will make it easier to disable this horrible behavior down the road.